### ISSUE TITLE
Why `max-age=0` for `no-cache` mode ?

### ISSUE THREAD

**Issue by Jxck** [whatwg/fetch#1121]
https://fetch.spec.whatwg.org/#http-network-or-cache-fetch

> 13. If httpRequest’s cache mode is "no-cache", httpRequest’s prevent no-cache cache-control header modification flag is unset, and httpRequest’s header list does not contain `Cache-Control`, then append `Cache-Control`/`max-age=0` to httpRequest’s header list.

I don't understand why not use `Cache-Control: no-cache` but use  `Cache-Control: max-age=0` for no-cache mode ?
I think this change happens in #232 but it mentions Gecko's behavior only and no reason why to do that.
Cache-Control is HTTP/1.1 so it reasonable using Pragma for HTTP/1.0, but `no-cache` and `max-age` is in the same spec so it seems that there are no reason to use `no-cache` instead of `max-age=0` for me.

cc. @annevk @ehsan

**Comment by annevk** [comment]
What do implementations do?

**Comment by Jxck** [comment]
As far as I know, Chrome/Safari/Firefox seems respect this spec.

- reload: max-age=0 with conditional headers
- force reload: no-cache without conditional headers

**Comment by asutherland** [comment]
It seems like there was additional discussion of this in https://github.com/whatwg/fetch/issues/722 too

**Comment by Jxck** [comment]
@mnot do you have any idea about here ?

**Comment by mnot** [comment]
I agree that *if* the intent is to prevent downstream caches from using a stored response, `no-cache` would be a better choice -- in particular because it includes the semantics of `must-revalidate`. 

But, as I argue in  #722, that's not the right thing to do -- Fetch setting *any* request cache-control headers is layering in a higher-level, API-specific semantic, not allowing developers to manipulate lower-layer HTTP semantics. Developers already have direct control of the headers, and setting them automatically deprives developers from the ability to control the local cache separately from remote ones.

**Comment by Jxck** [comment]
@mnot Thanks, I'm agree with `fetch()` API design.
And also browser seems implement "reload" UI based on this spec.
I don't know why firefox originally sends `max-age=0` when reloading, but if there are no appropriate reason to do that it's better to update the spec using `no-cache`.
and if there are an appropriate reason (maybe compatibility ?), it seems better to be noted in spec.

(my first question is "why browser sending max-age=0 when reloading" and reached to this spec)
